src/mainviews/create-job.tsx

```diff
@@ -160,22 +167,32 @@ export function CreateJob(props: ICreateJobProps): JSX.Element {
       const envObj = environmentList.find(env => env.name === target.value);
       // Validate that the default compute type is in fact in the list
       let newComputeType = envObj?.compute_types?.[0];
+
       if (
         envObj?.default_compute_type &&
         envObj?.compute_types &&
         envObj?.compute_types.includes(envObj?.default_compute_type)
       ) {
         newComputeType = envObj.default_compute_type;
       }
+
       const newEnvOutputFormats = outputFormatsForEnvironment(
         environmentList,
         target.value
       )?.map(format => format.name);
+
+      // Does the new environment support time zones?
+      let newTimeZone = 'UTC';
+      if (!envObj?.utc_only) {
+        newTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+      }
+
       props.handleModelChange({
         ...props.model,
         environment: target.value,
         computeType: newComputeType,
-        outputFormats: newEnvOutputFormats
+        outputFormats: newEnvOutputFormats,
+        timezone: newTimeZone
```

Comment:
The problem with doing this on select is that it resets the timezone even if you're switching between environments that support timezones. For example: browser's current timezone is 'X'. User sets to 'Y'. User then changes environments, and the timezone is always reset to 'X' even when it doesn't need to be.